tests/telemetry/test_instrumentation.py

```diff
@@ -160,3 +165,177 @@ def test_logger_provider(
     log_handler = get_log_handler()
     assert isinstance(log_handler, LoggingHandler)
     assert log_handler._logger_provider == logger_provider
+
+
+class TestTaskRunInstrumentation:
+    def assert_has_attributes(self, span, **expected_attributes):
+        """Helper method to assert span has expected attributes"""
+        for key, value in expected_attributes.items():
+            assert (
+                span.attributes.get(key) == value
+            ), f"Expected {key}={value}, got {span.attributes.get(key)}"
+
+    @pytest.fixture
+    async def task_run_engine(self, instrumentation):
+        @task
+        async def test_task(x: int, y: int):
+            return x + y
+
+        task_run = TaskRun(
+            id=uuid4(),
+            task_key="test_task",
+            flow_run_id=uuid4(),
+            state=Running(),
+            dynamic_key="test_task-1",
+        )
+
+        engine = AsyncTaskRunEngine(
+            task=test_task,
+            task_run=task_run,
+            parameters={"x": 1, "y": 2},
```

Comment:
Looks like all these tests only use the `AsyncTaskRunEngine`. Can you add tests to ensure the `SyncTaskRunEngine` also works with otel as exepcted.